### PR TITLE
Replace Travis with GitHub Actions

### PR DESCRIPTION
**Pull request by asbjornu**
Replace Travis with GitHub Actions

### CODE REVIEW

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412&bid=17208496)

File: tests/bootstrap.php [hunk 1]
```diff
@@ -29,4 +30,11 @@
 
     define('MERCHANT_TOKEN', $config['merchant_token']);
     define('PAYEE_ID', $config['payee_id']);
+
+    $data = json_decode(file_get_contents(__DIR__ . '../composer.json'), true);
```
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Client.php [hunk 2]
```diff
@@ -62,8 +62,25 @@ public function __construct($data = [])
...
         // Set version
         if (!$this->offsetExists(self::CLIENT_VERSION)) {
-            $data = json_decode(file_get_contents(__DIR__ . '/../../../../composer.json'), true);
-            $this->setVersion($data['version']);
+            $version = false;
+
+            if (defined('VERSION')) {
+                $version = VERSION;
+            } elseif (getenv("VERSION") !== false) {
+                $version = getenv("VERSION");
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Client.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Client.php [hunk 3]
```diff
@@ -62,8 +62,25 @@ public function __construct($data = [])
...
+            } else {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Client.php]: Codacy Issue found: [The method __construct uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php [hunk 4]
```diff
@@ -16,9 +16,10 @@
 
 require_once __DIR__ . '/TestCase.php';
 
-if (getenv('MERCHANT_TOKEN') && getenv('PAYEE_ID')) {
+if (getenv('MERCHANT_TOKEN') && getenv('PAYEE_ID') && getenv('VERSION')) {
     define('MERCHANT_TOKEN', getenv('MERCHANT_TOKEN'));
-    define('PAYEE_ID', getenv('PAYEE_ID'));        
+    define('PAYEE_ID', getenv('PAYEE_ID'));
+    define('VERSION', getenv('VERSION'));
```
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Client.php [hunk 5]
```diff
@@ -62,8 +62,25 @@ public function __construct($data = [])
 
         // Set version
         if (!$this->offsetExists(self::CLIENT_VERSION)) {
-            $data = json_decode(file_get_contents(__DIR__ . '/../../../../composer.json'), true);
-            $this->setVersion($data['version']);
+            $version = false;
+
+            if (defined('VERSION')) {
+                $version = VERSION;
+            } elseif (getenv("VERSION") !== false) {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Client.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 10
           

Complexity increasing per file
==============================
- src/SwedbankPay/Api/Client/Version.php  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412&bid=17208496)

File: src/SwedbankPay/Api/Client/Version.php [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    {
+        $version = false;
+
+        if (defined('VERSION')) {
+            $version = VERSION;
+        } elseif (getenv("VERSION") !== false) {
+            $version = getenv("VERSION");
+        } else {
+            $composer_json_path = __DIR__ . '/../../../../composer.json';
+            $composer_json_contents = file_get_contents(composer_json_path);
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [Avoid unused local variables such as '$composer_json_contents'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php (hunk 4 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+    public static function getVersion()
+    {
+        $version = false;
+
+        if (defined('VERSION')) {
+            $version = VERSION;
+        } elseif (getenv("VERSION") !== false) {
+            $version = getenv("VERSION");
+        } else {
+            $composer_json_path = __DIR__ . '/../../../../composer.json';
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [Avoid unused local variables such as '$composer_json_path'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+     * @return string Version number
+     * @throws ClientException
+     */
+    public static function getVersion()
+    {
+        $version = false;
+
+        if (defined('VERSION')) {
+            $version = VERSION;
+        } elseif (getenv("VERSION") !== false) {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php (hunk 6 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [Avoid excessively long variable names like $composer_json_contents. Keep variable name length under 20.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php (hunk 1 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+            $version = getenv("VERSION");
+        } else {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The method getVersion uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 10]
```diff
@@ -0,0 +1,42 @@
...
+     * @throws ClientException
+     */
+    public static function getVersion()
+    {
+        $version = false;
+
+        if (defined('VERSION')) {
+            $version = VERSION;
+        } elseif (getenv("VERSION") !== false) {
+            $version = getenv("VERSION");
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php (hunk 6 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Client.php [hunk 11]
```diff
@@ -62,8 +62,8 @@ public function __construct($data = [])
 
         // Set version
         if (!$this->offsetExists(self::CLIENT_VERSION)) {
-            $data = json_decode(file_get_contents(__DIR__ . '/../../../../composer.json'), true);
-            $this->setVersion($data['version']);
+            $version = Version::getVersion();
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Client.php]: Codacy Issue found: [Avoid using static access to class 'SwedbankPay\Api\Client\Version' in method '__construct'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 8
           

Complexity increasing per file
==============================
- src/SwedbankPay/Api/Client/Version.php  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412&bid=17208496)

File: tests/bootstrap.php (hunk 1 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php (hunk 4 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php (hunk 10 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Client.php (hunk 11 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Client.php]: Codacy Issue found: [Avoid using static access to class 'SwedbankPay\Api\Client\Version' in method '__construct'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php (hunk 9 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The method getVersion uses an else expression. Else clauses are basically not necessary and you can simplify the code by not using them.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+        } else {
+            $composer_json_path = __DIR__ . '/../../../../composer.json';
+            $composer_json_contents = file_get_contents($composer_json_path);
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [Avoid excessively long variable names like $composer_json_contents. Keep variable name length under 20.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php (hunk 8 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 8
           

Complexity increasing per file
==============================
- src/SwedbankPay/Api/Client/Version.php  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412&bid=17208496)

File: tests/bootstrap.php (hunk 1 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Client.php (hunk 11 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Client.php]: Codacy Issue found: [Avoid using static access to class 'SwedbankPay\Api\Client\Version' in method '__construct'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+        if (defined('VERSION')) {
+            return VERSION;
+        }
+
+        if (getenv("VERSION") !== false) {
+            return getenv("VERSION");
+        }
+
+        $composer_json_path = __DIR__ . '/../../../../composer.json';
+        $composer_json_contents = file_get_contents($composer_json_path);
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [Avoid excessively long variable names like $composer_json_contents. Keep variable name length under 20.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 14]
```diff
@@ -0,0 +1,41 @@
...
+     * @SuppressWarnings(PHPMD.ElseExpression)
+     */
+    public static function getVersion()
+    {
+        if (defined('VERSION')) {
+            return VERSION;
+        }
+
+        if (getenv("VERSION") !== false) {
+            return getenv("VERSION");
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 15]
```diff
@@ -0,0 +1,41 @@
...
+
+        $composer_json_path = __DIR__ . '/../../../../composer.json';
+        $composer_json_contents = file_get_contents($composer_json_path);
+        $data = json_decode($composer_json_contents, true);
+
+        if (isset($data['version'])) {
+            return $data['version'];
+        }
+
+        if ($version === false) {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [Avoid unused local variables such as '$version'.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php (hunk 4 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php (hunk 13 above)
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 16]
```diff
@@ -0,0 +1,41 @@
...
+     * @throws ClientException
+     * @SuppressWarnings(PHPMD.ElseExpression)
+     */
+    public static function getVersion()
+    {
+        if (defined('VERSION')) {
+            return VERSION;
+        }
+
+        if (getenv("VERSION") !== false) {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

**Comment by payex-dev** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
+ Solved 1
- Added 5
           

Complexity increasing per file
==============================
- src/SwedbankPay/Api/Client/Version.php  4
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412&bid=17208496)

File: src/SwedbankPay/Api/Client/Version.php [hunk 17]
```diff
@@ -0,0 +1,61 @@
...
+     * @return string Version number
+     * @throws ClientException
+     */
+    protected function getVersionFromEnvironment()
+    {
+        if (defined('VERSION')) {
+            return VERSION;
+        }
+
+        if (getenv("VERSION") !== false) {
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 18]
```diff
@@ -0,0 +1,61 @@
...
+            return getenv("VERSION");
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: src/SwedbankPay/Api/Client/Version.php [hunk 19]
```diff
@@ -0,0 +1,61 @@
...
+        }
+
+        $path = __DIR__ . '/../../../../composer.json';
+        $contents = file_get_contents($path);
```
Comment (payex-dev) [inline src/SwedbankPay/Api/Client/Version.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php (hunk 4 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function getenv() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

File: tests/bootstrap.php (hunk 1 above)
Comment (payex-dev) [inline tests/bootstrap.php]: Codacy Issue found: [The use of function file_get_contents() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-sdk-php/pullRequest?prid=5251412)

**Comment by asbjornu** [comment]
I had to do some work around how `version` is picked up from `composer.json`, so I would appreciate a quick code review, @aait @olegisk.

**Pull request merged**